internal/config/config.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"sigs.k8s.io/kubebuilder/pkg/model/config"
+	"sigs.k8s.io/yaml"
+)
+
+// Default path for the configuration file
+const DefaultPath = "PROJECT"
+
+func exists(path string) (bool, error) {
```

Comment:
I will have a look. It doesn't make that much sense to me either, it may be inherited from the old code.

Reply:
So the issue is that we are using `os.Stat` to discover if the file exists, and it may yield errors which are os-dependant. I added the `os.IsExist(err)` condition and I was already handling `err == nil` and `os.IsNotExist(err)`. So those three cases will return `true || false, nil`. But as `os.Stat` is os-dependant, we can not ensure that `err` falls into one of those three categories and thus the `Exists` check failed. I was trying to think of errors that may not fall under those 3 categories (that should be pretty rare) and maybe a directory permission could be one of them, as we would not be able to see if the file exists.